### PR TITLE
Enforce passing by keyword argument in `QNode` and `qml.execute`

### PR DESCRIPTION
**Pull request by albi3ro**
**Context:**

Passing too many arguments positionally is error prone.  It can be difficult to remember what the proper order of the arguements is, and it can be easy to reorder keyword arguments.

**Description of the Change:**

Start mandating that various keyword arguments *are* passed as keyword arguments in `QNode` and `qml.execute`

**Benefits:**

We forbid entire classes of mistakes.

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -37,6 +37,8 @@
   visualizations, allowing global and per-wire customization with options like `color`, `linestyle`, and `linewidth`.
   [(#6486)](https://github.com/PennyLaneAI/pennylane/pull/6486)
 
+* `QNode` and `qml.execute` now forbid certain keyword arguments from being passed positionally.
```
Comment (albi3ro) [inline doc/releases/changelog-dev.md:40]:
```suggestion
* `QNode` and `qml.execute` now forbid certain keyword arguments from being passed positionally.
  [(#6610)](https://github.com/PennyLaneAI/pennylane/pull/6610)
```

**Review by andrijapau — APPROVED** [review]
Looks good to me! Nice clean-up 🚀

**Review by PietropaoloFrisoni — APPROVED** [review]

File: doc/releases/changelog-dev.md [hunk 2]
```diff
@@ -156,6 +159,10 @@
 
 <h3>Breaking changes 💔</h3>
 
+* `QNode` and `qml.execute` now forbid certain keyword arguments from being passed positionally.
+  This helps prevent keyword arguments from being passed in the wrong order.
+  [(#6610)](https://github.com/PennyLaneAI/pennylane/pull/6610)
+
```
Comment (mudit2812) [inline doc/releases/changelog-dev.md:165]: Is it intentional to have 2 changelog entries?

Comment (albi3ro) [reply doc/releases/changelog-dev.md:165]: improvement versus breaking change?

Comment (albi3ro) [reply doc/releases/changelog-dev.md:165]: But actually, this isn't really a breaking change of importance. I'll remove it from here.

**Comment by mudit2812** [comment]
I'm excited to find out how many tests will need to be updated 😅

File: doc/releases/changelog-dev.md (hunk 2 above)
Comment (albi3ro) [inline doc/releases/changelog-dev.md:165]:
```suggestion
```

**Comment by albi3ro** [comment]
> I'm excited to find out how many tests will need to be updated 😅

exactly zero 😆

**Pull request merged**